File: client/coresource/src/ru/spbstu/gyboml/core/net/Network.java
package ru.spbstu.gyboml.core.net;

import com.esotericsoftware.kryo.Kryo;
import com.esotericsoftware.kryonet.EndPoint;

import java.util.ArrayList;
import java.util.List;

import ru.spbstu.gyboml.core.Player;
import ru.spbstu.gyboml.core.PlayerType;
import ru.spbstu.gyboml.core.net.SessionRequests.ExitSession;
import ru.spbstu.gyboml.core.net.SessionRequests.GetSessions;
import ru.spbstu.gyboml.core.net.SessionRequests.Ready;


/*
 * Common network settings between client and server.
 */
public class Network {

    // server ports
    public static final int tcpPort = 4445;
    public static final int udpPort = 3335;

    // server address
    public static final String serverAddress = "34.91.65.96";

    // register request/response type for kryo serialization
    public static void register(EndPoint endPoint) {
        Kryo kryo = endPoint.getKryo();
        kryo.register(String[].class);
        kryo.register(List.class);
        kryo.register(ArrayList.class);
        kryo.register(Player.class);
        kryo.register(PlayerType.class);
        kryo.register(Integer.class);
        kryo.register(Float.class);

        kryo.register(SessionRequests.RegisterName.class);
        kryo.register(SessionRequests.ConnectSession.class);
        kryo.register(SessionRequests.CreateSession.class);
        kryo.register(SessionInfo.class);
        kryo.register(GetSessions.class);
        kryo.register(Ready.class);
        kryo.register(ExitSession.class);
        kryo.register(ExitSession.class);
        kryo.register(GameRequests.GameExit.class);
        kryo.register(GameRequests.Shoot.class);

        kryo.register(SessionResponses.ReadyApproved.class);
        kryo.register(SessionResponses.SessionExited.class);
        kryo.register(SessionResponses.SessionCreated.class);
        kryo.register(SessionResponses.SessionConnected.class);
        kryo.register(SessionResponses.ServerError.class);
        kryo.register(SessionResponses.TakeSessions.class);
        kryo.register(SessionResponses.SessionStarted.class);
        kryo.register(GameResponses.Shooted.class);
        kryo.register(GameResponses.GameExited.class);
        kryo.register(GameResponses.PassTurned.class);
    }
}

File: client/coresource/src/ru/spbstu/gyboml/core/net/Network.kt
package ru.spbstu.gyboml.core.net

import com.badlogic.gdx.math.Vector2
import com.esotericsoftware.kryonet.EndPoint
import ru.spbstu.gyboml.core.Player
import ru.spbstu.gyboml.core.PlayerType
import ru.spbstu.gyboml.core.destructible.Material
import ru.spbstu.gyboml.core.net.SessionRequests.*
import ru.spbstu.gyboml.core.shot.ShotType
import java.util.*

object Network {
    const val tcpPort = 4445
    const val udpPort = 3335

    const val address = "192.168.1.177"//"34.91.65.96"

    fun register(endpoint: EndPoint) {
        with (endpoint.kryo) {
            register(Array<String>::class.java)
            register(List::class.java)
            register(ArrayList::class.java)
            register(Player::class.java)
            register(PlayerType::class.java)
            register(Integer::class.java)
            register(Float::class.java)
            register(Vector2::class.java)
            register(SessionPlayer::class.java)

            register(SessionRequests.RegisterName::class.java)
            register(SessionRequests.ConnectSession::class.java)
            register(SessionRequests.CreateSession::class.java)
            register(SessionInfo::class.java)
            register(GetSessions::class.java)
            register(Ready::class.java)
            register(ExitSession::class.java)
            register(ExitSession::class.java)
            register(GameRequests.GameExit::class.java)
            register(GameRequests.GameLoaded::class.java)

            register(SessionResponses.ReadyApproved::class.java)
            register(SessionResponses.SessionExited::class.java)
            register(SessionResponses.SessionCreated::class.java)
            register(SessionResponses.SessionConnected::class.java)
            register(SessionResponses.TakeSessions::class.java)
            register(SessionResponses.NameRegistred::class.java)
            register(SessionResponses.SessionStarted::class.java)
            register(GameResponses.GameExited::class.java)
            register(GameResponses.PassTurned::class.java)
            register(GameResponses.GameStarted::class.java)

            register(GameMessage.CreateShot::class.java)
            register(GameMessage.UpdateShot::class.java)
            register(GameMessage.RemoveShot::class.java)

            register(GameMessage.CreateBlock::class.java)
            register(GameMessage.UpdateBlock::class.java)
            register(GameMessage.RemoveBlock::class.java)

            register(Material::class.java)
            register(ShotType::class.java)
        }
    }
}